test/e2e/tls_test.go

```diff
@@ -0,0 +1,517 @@
+	if err != nil {
+		t.Log("====Error: ", err)
+	}
+
+	err = GenerateCertificates("openssl genpkey -algorithm RSA -out ../test_data/pod_one_certs/tls.key")
+	if err != nil {
+		t.Log("====Error: ", err)
+	}
+
+	err = GenerateCertificates("openssl req -new -key ../test_data/pod_one_certs/tls.key -config ../test_data/pod_one_certs/server.cnf -out ../test_data/pod_one_certs/tls.csr")
+	if err != nil {
+		t.Log("====Error: ", err)
+	}
+
+	err = GenerateCertificates("openssl x509 -req -CA ../test_data/ca_certs/cacert.pem -CAkey ../test_data/ca_certs/ca-private-key.pem -CAcreateserial -CAserial ../test_data/pod_one_certs/cacert.srl -in ../test_data/pod_one_certs/tls.csr -out ../test_data/pod_one_certs/tls.crt -days 365")
+	if err != nil {
+		t.Log("====Error: ", err)
+	}
+
+	// create secret for ca certificate
+	t.Logf("====Creating secret for ca certificate")
+	k8s.RunKubectl(t, kubectlOptions, "create", "secret", "generic", "ca-cert", "--from-file=../test_data/ca_certs/cacert.pem")
+
+	// create secret for named certificate for pod-0
+	t.Logf("====Creating secret for pod-0 certificate")
+	k8s.RunKubectl(t, kubectlOptions, "create", "secret", "generic", "marklogic-0-cert", "--from-file=../test_data/pod_zero_certs/tls.crt", "--from-file=../test_data/pod_zero_certs/tls.key")
+
+	// create secret for named certificate for pod-1
+	t.Logf("====Creating secret for pod-1 certificate")
+	k8s.RunKubectl(t, kubectlOptions, "create", "secret", "generic", "marklogic-1-cert", "--from-file=../test_data/pod_one_certs/tls.crt", "--from-file=../test_data/pod_one_certs/tls.key")
+
+	t.Logf("====Installing Helm Chart")
+	helm.Install(t, options, helmChartPath, releaseName)
+
+	podName := releaseName + "-0"
+	podOneName := releaseName + "-1"
+
+	tlsConfig := tls.Config{InsecureSkipVerify: true}
+
+	// wait until the pod is in Ready status
+	k8s.WaitUntilPodAvailable(t, kubectlOptions, podName, 10, 20*time.Second)
+	tunnel7997 := k8s.NewTunnel(kubectlOptions, k8s.ResourceTypePod, podName, 7997, 7997)
+	defer tunnel7997.Close()
+	tunnel7997.ForwardPort(t)
+	endpoint7997 := fmt.Sprintf("http://%s", tunnel7997.Endpoint())
+
+	// verify if 7997 health check endpoint returns 200
+	http_helper.HttpGetWithRetryWithCustomValidation(
+		t,
+		endpoint7997,
+		&tlsConfig,
+		10,
+		15*time.Second,
+		func(statusCode int, body string) bool {
+			return statusCode == 200
+		},
+	)
+
+	// wait until pods are in Ready status
+	k8s.WaitUntilPodAvailable(t, kubectlOptions, podName, 15, 30*time.Second)
+	k8s.WaitUntilPodAvailable(t, kubectlOptions, podOneName, 15, 30*time.Second)
+
+	// get corev1.Pod to get logs of a pod
+	pod := k8s.GetPod(t, kubectlOptions, podName)
+
+	// get pod logs to verify named certificate is used
+	t.Logf("====Getting pod logs")
+	podLogs := k8s.GetPodLogs(t, kubectlOptions, pod, "")
+
+	// verify logs if named certificate is used
+	if !strings.Contains(podLogs, "Info: [poststart] certType in postStart: named") {
```

Comment:
We should not use internal state like log message to verify if HTTPS is turned on successfully. 
First the log message could change in the future and it cause the false negative test result. Second if there are bugs in the code and HTTPs, just reading the log message won't be able to detect such failure, causing the false positive test result.

I think we should use the HTTP Client to connect to the default end point to make sure it's using HTTP connection. And then send a rest call fetch the status of the defualtTemplate to check if the cert is using temporary cert or named cert.

Reply:
updated tests to use HTTP calls to verify status of the named certificate and defualtTemplate